cli/internal/state/state_test.go

```diff
@@ -326,3 +327,80 @@ func TestMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestMarshalHexBytes(t *testing.T) {
```

Comment:
nit: I would test marshal and unmarshal together to verify that they are "compatible"